src/ui/note/mod.rs

```diff
@@ -415,6 +455,36 @@ impl<'a> NoteView<'a> {
     }
 }
 
+fn get_reposted_note_key(ndb: &Ndb, txn: &Transaction, note: &Note) -> Option<NoteKey> {
+    let new_note_id: &[u8; 32] = if note.kind() == 6 {
+        let mut res = None;
+        for tag in note.tags().iter() {
```

Comment:
```suggestion
        for tag in note.tags().iter() {
            if tag.count() == 0 {
	        continue;
            }
```

this is needed or we can crash on `unwrap` or `get_unchecked` with a malformed note, but maybe we should just do this so its safe even during refactors:

```rust
if let Some("e") = tag.get(0).and_then(|t|t.variant().str())
```